tools/Google.Cloud.Tools.TrimCoverageReport/Program.cs

```diff
@@ -109,13 +115,61 @@ private static void RecomputeAndSimplify(XDocument document, string parentElemen
         private static (int statements, int covered) GetContainerCoverage(XElement container) =>
             ((int)container.Attribute(TotalStatements), (int)container.Attribute(CoveredStatements));
 
+        private static void FixPaths(IEnumerable<XElement> fileElements)
+        {
+            if (!s_fixFilePaths) return;
+            foreach (var fileNode in fileElements)
+            {
+                fileNode.Attribute("Name").Value = AdjustedLocalPath(fileNode.Attribute("Name")?.Value);
+                Console.WriteLine(fileNode.Attribute("Name").Value);
+            }
+        }
+
+        // If for some reason we are running on a new git clone with a different full path, we'll still
+        // try to fix up the stored path by looking for a relative path under the current working directory.
+        private static string AdjustedLocalPath(string originalPath)
+        {
+            if (string.IsNullOrEmpty(originalPath))
+            {
+                return originalPath;
+            }
+            var originalFile = new FileInfo(originalPath);
+            if (originalFile.Exists)
+            {
+                return originalPath;
+            }
+
+            var currentFolder = originalFile.Directory;
+            while (currentFolder != null)
+            {
+                var relativePath = originalFile.FullName.Substring(currentFolder.FullName.Length + 1);
+                var testFile = Path.Combine(Directory.GetCurrentDirectory(), relativePath);
+                if (File.Exists(testFile))
+                {
+                    return testFile;
+                }
+                currentFolder = currentFolder.Parent;
+            }
+            return originalPath;
+        }
+
         /// <summary>
         /// Detects whether a file is generated. All three code generators we use (protoc, grpc,
         /// toolkit) include a comment starting with "// Generated" early in the file.
         /// </summary>
         /// <param name="path"></param>
         /// <returns></returns>
-        static bool IsGeneratedFile(string path) =>
-            File.ReadLines(path).Take(20).Any(line => line.StartsWith("// Generated"));
+        private static bool IsGeneratedFile(string path)
```

Comment:
I'd use `File.Exists` here instead:

```csharp
return (s_skipBadPaths && !File.Exists(path)) || (code as before)
```

I'd also rename the method. (I very nearly had this code in the original version...)